src/hyperion/experiment_plans/grid_detect_then_xray_centre_plan.py

```diff
@@ -219,6 +221,7 @@ def run_grid_detection_plan(
         panda=composite.panda,
         fast_grid_scan=composite.fast_grid_scan,
         dcm=composite.dcm,
+        robot=composite.robot,
```

Comment:
I think it might be best to accept duck-typing for our composites and just to do `typing.cast()`.

Reply:
Yh, I saw you did this in the `robot_load` and was a bit on the fence of on it. Certainly makes the code cleaner but introduces potential errors where the device is named differently in the different composites. @dperl-dls and @olliesilvester, what are your thoughts?

Reply:
Haven't got a strong opinion, but we could make sure the devices are always named the same way across composites and then use `typing.cast()`